src/Ztm.WebApi/TransactionConfirmationWatcher.cs

```
@@ -165,17 +166,30 @@ async Task SetupTimerAsync(Guid id)
 
             this.timerLock.EnterWriteLock();
 
-            var watch = await this.watchRepository.GetAsync(id, CancellationToken.None);
             try
             {
-                if (!timers.ContainsKey(watch.Transaction))
+                var watch = await this.watchRepository.GetAsync(id, CancellationToken.None);
+
+                Dictionary<Guid, Tuple<Timer, int>> timers;
+
+                if (!this.timers.TryGetValue(watch.Transaction, out timers))
                 {
-                    timers[watch.Transaction] = new Dictionary<Guid, Tuple<Timer, int>>();
+                    timers = new Dictionary<Guid, Tuple<Timer, int>>();
+                    this.timers.Add(watch.Transaction, timers);
                 }
 
-                timers[watch.Transaction][watch.Id] = new Tuple<Timer, int>(timer, watch.Confirmation);
-                timer.Elapsed += OnTimeout;
-                timer.Start(watch.RemainingWaitingTime < TimeSpan.Zero ? TimeSpan.Zero : watch.RemainingWaitingTime, null, watch.Id);
+                timers.Add(watch.Id, new Tuple<Timer, int>(timer, watch.Confirmation));
+
+                try
+                {
+                    timer.Elapsed += OnTimeout;
+                    timer.Start(watch.RemainingWaitingTime < TimeSpan.Zero ? TimeSpan.Zero : watch.RemainingWaitingTime, null, watch.Id);
+                }
+                catch (Exception)
```

Just `catch` is enough to catch all exception.